client/consensus/src/lib.rs

```diff
@@ -149,6 +154,22 @@ impl<B, I, C> BlockImport<B> for FrontierBlockImport<B, I, C> where
 					}
 				},
 			}
+			// On importing block 1 we also map the genesis block in the auxiliary.
```

Comment:
This approach of doing the mapping has the one disadvantage that the node won't be able to answer queries about the genesis block (by hash) until it has imported block 1. That being noted, this approach is certainly good enough in practice, and is the only way I see to do it from the import pipeline.

One maybe workable alternative I see is putting the ethereum genesis block hash in the chain spec's [extension field](https://substrate.dev/docs/en/knowledgebase/integrate/chain-spec#extension), and having the node write it to aux storage when it starts a new chain. This alternative path has some uncertainty and added complexity for very little practical payoff, so I support keeping it in the import pipeline.

Reply:
> This approach of doing the mapping has the one disadvantage that the node won't be able to answer queries about the genesis block (by hash) until it has imported block 1. 

Yes, unfortunately that's the case.

> One maybe workable alternative I see is putting the ethereum genesis block hash in the chain spec's extension field, and having the node write it to aux storage when it starts a new chain. 

Seems a more natural alternative than the current proposal tbh, but as you said the inmediate benefit we gain from it is almost nothing. Although I'd be very interested in how to use the extension field :)